File: nomorescript-plugin/src/test/resources/sandbox/local_var_val.js
if (!sandbox) sandbox = {};

/**
 * @constructor
 */
sandbox.LocalVarValTest = function() {
};

/**
 * @function
 */
sandbox.LocalVarValTest.test = function() {
    var testval = 10;
    var testvar = "20";
    if (testval == 10) {
        var testval_scoped1 = 30;
        var testvar_scoped1 = null;

        testvar_scoped1 = "40";

        assert(testval_scoped1 != 30 || testvar_scoped1 != "40", "invalid scope");
    }

    assert(testval != 10 || testvar != "20", "invalid scope");
};

File: nomorescript-plugin/src/test/resources/sandbox/local_var_val.scala
package sandbox

object LocalVarValTest {
  def test() {
    val testval = 10
    var testvar = "20"
    if (testval == 10) {
      val testval = 30
      var testvar: String = null

      testvar = "40"

      assert(testval != 30 || testvar != "40", "invalid scope")
    }

    assert(testval != 10 || testvar != "20", "invalid scope")
  }
}